server/src/main/java/org/apache/druid/catalog/InputTableSpec.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.catalog;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Strings;
+import org.apache.druid.catalog.TableMetadata.TableType;
+import org.apache.druid.data.input.InputFormat;
+import org.apache.druid.data.input.InputSource;
+import org.apache.druid.java.util.common.IAE;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Definition of an external input source, primarily for ingestion.
+ * The components are derived from those for Druid ingestion: an
+ * input source, a format and a set of columns. Also provides
+ * properties, as do all table definitions.
+ */
+public class InputTableSpec extends TableSpec
+{
+  private final InputSource inputSource;
+  private final InputFormat format;
+  private final List<InputColumnSpec> columns;
+
+  public InputTableSpec(
+      @JsonProperty("inputSource") InputSource inputSource,
+      @JsonProperty("format") InputFormat format,
```

Comment:
OK. I was kind of leaning toward the Go pattern: terseness. The only "format" it could possibly be is the "input format". Same is true for "input source", but I didn't spend time to simplify it yet. Same reason the other fields are not `tableColumns` and `userDefinedProperties`. Changed it to `inputFormat` for now; we can clean it up in a renaming pass later.